src/EventStore.Core.Tests/Services/Transport/Tcp/TcpClientDispatcherTests.cs

```diff
@@ -0,0 +1,224 @@
+using EventStore.Core.Bus;
+using EventStore.Core.Services.Transport.Tcp;
+using NUnit.Framework;
+using System;
+using EventStore.Core.Authentication;
+using EventStore.Core.Messaging;
+using EventStore.Core.Tests.Authentication;
+using System.Linq;
+using EventStore.Core.Data;
+using EventStore.Core.Messages;
+using System.Text;
+using EventStore.Core.Services.UserManagement;
+using EventStore.Core.TransactionLog.LogRecords;
+using EventStore.Core.Services;
+
+namespace EventStore.Core.Tests.Services.Transport.Tcp
+{
+    [TestFixture]
+    public class TcpClientDispatcherTests
+    {
+        private readonly NoopEnvelope _envelope = new NoopEnvelope();
+        private const byte _version = (byte)ClientVersion.V1;
+
+        private ClientTcpDispatcher _dispatcher;
+        private TcpConnectionManager _connection;
+
+        [OneTimeSetUp]
+        public void Setup()
+        {
+            _dispatcher = new ClientTcpDispatcher();
+
+            var dummyConnection = new DummyTcpConnection();
+            _connection = new TcpConnectionManager(
+                Guid.NewGuid().ToString(), TcpServiceType.External, new ClientTcpDispatcher(),
+                InMemoryBus.CreateTest(), dummyConnection, InMemoryBus.CreateTest(), new InternalAuthenticationProvider(
+                                new Core.Helpers.IODispatcher(InMemoryBus.CreateTest(), new NoopEnvelope()), new StubPasswordHashAlgorithm(), 1),
+                TimeSpan.FromSeconds(10), TimeSpan.FromSeconds(10), (man, err) => { });
+        }
+
+        [Test]
+        public void when_unwrapping_message_that_does_not_have_version1_unwrapper_should_use_version2_unwrapper()
+        {
+            var dto = new TcpClientMessageDto.DeleteStream("test-stream", 2, true, false);
+            var package = new TcpPackage(TcpCommand.DeleteStream, Guid.NewGuid(), dto.Serialize());
+
+            var msg = _dispatcher.UnwrapPackage(package, _envelope, SystemAccount.Principal, "", "", _connection, _version) as ClientMessage.DeleteStream;
+            Assert.IsNotNull(msg);
+        }
+
+        [Test]
+        public void when_wrapping_message_that_does_not_have_version1_wrapper_should_use_version2_wrapper()
+        {
+            var msg = new ClientMessage.DeleteStream(Guid.NewGuid(), Guid.NewGuid(), _envelope, true, "test-stream", 2, false, SystemAccount.Principal);
+            var package = _dispatcher.WrapMessage(msg, _version);
+            Assert.IsNotNull(package, "Package");
+            Assert.AreEqual(TcpCommand.DeleteStream, package.Value.Command);
+
+            var dto = package.Value.Data.Deserialize<TcpClientMessageDto.DeleteStream>();
+            Assert.IsNotNull(dto, "DTO");
```

Comment:
The descriptions for the failure cases should be consistent with the other failure cases in this PR. For example, "DTO is null". e.g. [here](https://github.com/EventStore/EventStore/pull/1235/commits/5b302e7b0edd704d99b7cfc21bab422767d5c5ee#diff-5ce4b7d447bba7a216eeaacad3c7e46aR77)